pkg/util/pod/pods.go

```diff
@@ -7,7 +7,8 @@ import (
 	"github.com/maistra/maistra-test-tool/pkg/util/test"
 )
 
-func MatchingSelector(selector string, ns string) ocpackage.PodLocatorFunc {
+// MatchingSelector returns a function that returns the name of a pod matching the given selector in the given namespace. If more than one pod matches the selector, will fail unless  returnOnePod is true, then the first pod is returned even if more than one pod matches the selector.
+func MatchingSelector(selector string, ns string, returnOnePod ...bool) ocpackage.PodLocatorFunc {
```

Comment:
No. This is an ugly hack. We need to fix this properly. 

Maybe by `MatchingSelector` returning all the pods it finds and then `Exec` throwing an error if multiple pods are received.

Reply:
Hummm ok ok, I'll change the behavior

Reply:
I've taken a closer look at this. I see you only want a single pod, even if multiple exist. In that case, I see two options:
- add an `options ...Option` parameter where you can specify what should happen if zero/one/multiple pods are found.
- add a new `MatchingSelectorFirst` or `MatchingSelectorAny` function that returns the first pod and ignores the rest. In this approach, this function and the existing `MatchingSelector` should of course call the same underlying function with a parameter that tells the function how to handle multiple matches).